code/controllers/subsystem/supply_shuttle.dm

```diff
@@ -298,14 +370,124 @@ var/datum/subsystem/supply_shuttle/SSsupply_shuttle
+		cargo_forward_cooldown = rand(CARGO_FORWARD_DELAY_MIN,CARGO_FORWARD_DELAY_MAX)
+
+		var/list/datum/cargo_forwarding/new_forwards = list()
+		if(prob(50) && previous_forwards_types && previous_forwards_types.len) // Keep it just a chance to get the previous round's forwards so we don't just end up with those
+			for(var/k in 1 to amount_forwarded)
+				if(!previous_forwards_types || !previous_forwards_types.len) // Break out if nothing sent
+					break
+				var/previous_index = rand(1,previous_forwards_types.len)
+				var/forwardtype = text2path(previous_forwards_types[previous_index])
+				var/datum/cargo_forwarding/CF = new forwardtype
+				new_forwards.Add(CF)
+				previous_forwards_types.Remove(previous_forwards_types[previous_index]) // Must be the index to remove a specific one
+				if(previous_forwards_stations && previous_forwards_stations.len)
+					var/index_to_pick = previous_index && previous_index < previous_forwards_stations.len ? previous_index : rand(1,previous_forwards_stations.len)
+					CF.origin_station_name = previous_forwards_stations[index_to_pick]
+					previous_forwards_stations.Remove(previous_forwards_stations[index_to_pick])
+				else if(!previous_forwards_types || !previous_forwards_types.len) // Cut the other fluff info list if the main type one is empty, to prevent desync with indices
+					previous_forwards_names.Cut()
+				if(previous_forwards_names && previous_forwards_names.len)
+					var/index_to_pick = previous_index && previous_index < previous_forwards_names.len ? previous_index : rand(1,previous_forwards_names.len)
+					CF.origin_sender_name = previous_forwards_names[index_to_pick]
+					previous_forwards_names.Remove(previous_forwards_names[index_to_pick])
+				else if(!previous_forwards_types || !previous_forwards_types.len)
+					previous_forwards_stations.Cut()
+				log_debug("CARGO FORWARDING: Persistence crate [forwardtype] loaded, from [CF.origin_sender_name] of [CF.origin_station_name].")
+		if(new_forwards.len < amount_forwarded) // If we got nothing or not the entire amount from the above
+			if(new_forwards.len)
+				log_debug("CARGO FORWARDING: [new_forwards.len] crates of [amount_forwarded] were persistence crates, now loading them as normal.")
+			for(var/j in 1 to (amount_forwarded - new_forwards.len))
+				if(prob(75)) // Normal orderable stuff
+					var/datum/cargo_forwarding/from_supplypack/SCF = new
+					new_forwards.Add(SCF)
+				else
+					if(prob(75)) // Centcomm orders
+						var/datum/cargo_forwarding/from_centcomm_order/OCF = new
+						new_forwards.Add(OCF)
+					else // Rare goodies not usually obtainable
+						var/cratetype = pick(subtypesof(/datum/cargo_forwarding/misc))
+						var/datum/cargo_forwarding/misc/MCF = new cratetype
+						new_forwards.Add(MCF)
+		
+		var/i = rand(1,clear_turfs.len)
+		var/turf/pickedloc = clear_turfs[i]
+		clear_turfs.Cut(i,i+1)
+		new /obj/machinery/crate_weigher(pickedloc)
```

Comment:
instead of spawning a weigher with every shipment, I think it would make more sense to just map one in to each cargo bay